filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/repository/exception/TokenErrorMessageWorker.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.filteredresource.repository.exception;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+
+import uk.gov.gchq.palisade.service.filteredresource.domain.TokenErrorMessageEntity;
+import uk.gov.gchq.palisade.service.filteredresource.repository.exception.TokenErrorMessageWorker.WorkerCommand;
+
+import java.util.List;
+
+/**
+ * A worker to carry-out a client request for a list of error messages (from kafka "error" topic or redis persistence)
+ * given a token (from websocket url "ws://filtered-resource-service/resource/$token")
+ */
+final class TokenErrorMessageWorker extends AbstractBehavior<WorkerCommand> {
+
+
+    protected interface WorkerCommand {
+        // Marker interface for inputs of the worker
+    }
+
+    protected interface WorkerResponse {
+        // Marker interface for outputs of the worker
+    }
+
+    /**
+     * A request to get all error messages for a unique token.
+     * The worker will {@link ActorRef#tell} the replyTo actor the error messages once found.
+     */
+    protected static class GetAllExceptions implements WorkerCommand {
+        protected final String token;
+        protected final ActorRef<WorkerResponse> replyTo;
+
+        protected GetAllExceptions(final String token, final ActorRef<WorkerResponse> replyTo) {
+            this.token = token;
+            this.replyTo = replyTo;
+        }
+    }
+
+    /**
+     * A response for this actor to send to its {@code replyTo} actor.
+     * This is received by the worker when appropriate error messages are found.
+     * This is both a possible input to the system {@link WorkerCommand} as well as an output {@link WorkerResponse}
+     */
+    protected static class SetAuditErrorMessages implements WorkerCommand, WorkerResponse {
```

Comment:
I think this is only an 'output' of the worker, never an 'input' (compared to offsets where a kafka message input tells the actor to return a value)
```suggestion
    protected static class SetAuditErrorMessages implements WorkerResponse {
```

Reply:
updated